Compilation error (CUDA, Linux), possibly XMR-STAK_THREADS

Trying to compile on Ubuntu 16.04, GTX 760, with the following options, I get an error (below). When I compile without setting `-DXMR-STAK_THREADS=32`, the compilation is successful and I get 320 H/s, using 64 threads, and 12 Blocks. Don't know, whether this is ok or not (first time user).

Compile options:
```
  -DMICROHTTPD_ENABLE=OFF
  -DMHTD=/usr/include/microhttpd.h
  -OpenSSL_ENABLE=ON
  -DXMR-STAK_CURRENCY=monero
  -DCPU_ENABLE=OFF
  -DHWLOC_ENABLE=OFF
  -DOpenCL_ENABLE=OFF
  -DCUDA_ENABLE=ON
  -DCUDA_ARCH=30
  -DXMR-STAK_LARGEGRID=OFF
  -DXMR-STAK_THREADS=32
```
Error:
```
[ 83%] Building NVCC (Device) object CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/xmrstak_cuda_backend_generated_cuda_core.cu.o
/home/ondrej/repo/xmr-stak/xmrstak/backend/nvidia/nvcc_code/cuda_core.cu(171): error: attributes are not allowed here

1 error detected in the compilation of "/tmp/tmpxft_0000514d_00000000-7_cuda_core.cpp1.ii".
CMake Error at xmrstak_cuda_backend_generated_cuda_core.cu.o.cmake:266 (message):
  Error generating file
  /home/ondrej/repo/xmr-stak/build/CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/./xmrstak_cuda_backend_generated_cuda_core.cu.o


CMakeFiles/xmrstak_cuda_backend.dir/build.make:63: recipe for target 'CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/xmrstak_cuda_backend_generated_cuda_core.cu.o' failed
make[2]: *** [CMakeFiles/xmrstak_cuda_backend.dir/xmrstak/backend/nvidia/nvcc_code/xmrstak_cuda_backend_generated_cuda_core.cu.o] Error 1
CMakeFiles/Makefile2:142: recipe for target 'CMakeFiles/xmrstak_cuda_backend.dir/all' failed
make[1]: *** [CMakeFiles/xmrstak_cuda_backend.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/xmrstak/backend/nvidia/nvcc_code/cuda_core.cu
+++ b/xmrstak/backend/nvidia/nvcc_code/cuda_core.cu
@@ -167,10 +167,10 @@ __forceinline__ __device__ uint32_t shuffle(volatile uint32_t* ptr,const uint32_
 #endif
 }
 
+template<size_t ITERATIONS, uint32_t THREAD_SHIFT, uint32_t MASK>
 #ifdef XMR_STAK_THREADS
 __launch_bounds__( XMR_STAK_THREADS * 4 )
 #endif
-template<size_t ITERATIONS, uint32_t THREAD_SHIFT, uint32_t MASK>
 __global__ void cryptonight_core_gpu_phase2( int threads, int bfactor, int partidx, uint32_t * d_long_state, uint32_t * d_ctx_a, uint32_t * d_ctx_b )
 {
 	__shared__ uint32_t sharedMemory[1024];

